src/game/common/system/mempoolobj.h

```diff
@@ -76,45 +76,44 @@ private: \
     virtual MemoryPool *Get_Object_Pool() override { return Get_Class_Pool(); } \
 \
 public: \
-    enum classname##MagicEnum{ classname##_GLUE_NOT_IMPLEMENTED }; \
+    enum classname##MagicEnum{ classname##_GLUE_NOT_IMPLEMENTED = 0 }; \
 \
-    void *operator new(size_t size); \
-    void *operator new(size_t size, classname##MagicEnum) \
+    void *operator new(size_t size, classname##MagicEnum = classname##_GLUE_NOT_IMPLEMENTED) \
     { \
         captainslog_dbgassert(size == sizeof(classname), \
             "The wrong operator new is being called; ensure all objects in the hierarchy have MemoryPoolGlue set up " \
             "correctly"); \
         return Get_Class_Pool()->Allocate_Block(); \
     } \
-    void operator delete(void *ptr) \
+    void operator delete(void *ptr) = delete; \
+    void operator delete(void *ptr, classname##MagicEnum) \
     { \
-        captainslog_debug("Please call deleteInstance instead of delete."); \
+        captainslog_dbgassert(0, "Please call Delete_Instance instead of delete."); \
```

Comment:
I promoted this to an `assert`, because really it should be treated as an error.